### PR TITLE
Refactored chrome.storage.sync.get calls in background

### PR DESCRIPTION
**Pull request by abhidas17695**
@vbanos I refactored background.js such that chrome.storage.sync.get is not called multiple times. I also solved the problem Max described ( asynchronous nature of chrome tabs/windows opening). Please review this PR

### CODE REVIEW

File: scripts/background.js [hunk 1]
```diff
@@ -173,235 +171,216 @@ chrome.webRequest.onCompleted.addListener(function(details) {
...
+          shouldOpen: event.tagcloud,
+          url: message.url
+        }
+      ];
       //If the Context is to be showed in tabs
-      if(event.show_context=="tab"){
-        if(tabId2==0 || tabId3==0 || tabId4==0 || tabId5==0 || tabId6==0 || tabId7==0 || tabId8==0 || tabId9 ==0){  //Checking if Tabs are not open already
+      if (event.show_context == "tab") {
+        if (tabId2 == 0 || tabId3 == 0 || tabId4 == 0 || tabId5 == 0 || tabId6 == 0 || tabId7 == 0 || tabId8 == 0 || tabId9 == 0) {  //Checking if Tabs are not open already
           //If not selected show-all option ,then check and open indivisually
```
Comment (vbanos) [inline scripts/background.js]: This comment is now irrelevant and should be removed.

**Comment by vbanos** [comment]
A general comment that applies everywhere: When checking for equals, please use `===` instead of `==`.
When checking for not equals, please use `!==` instead of `==`.

File: scripts/background.js [hunk 2]
```diff
@@ -13,31 +13,31 @@ var excluded_urls = [
...
-var windowIdSingle=0;
+var tabId2 = 0;
+var tabId3 = 0;
+var tabId4 = 0;
+var tabId5 = 0;
+var tabId6 = 0;
+var tabId7 = 0;
+var tabId8 = 0;
+var tabId9 = 0;
+var tabId10 = 0;
```
Comment (vbanos) [inline scripts/background.js]: We used to have 10 context windows but now we've got just 7.
The extra `tabId` and `windowId` vars need to be removed from everywhere (they are also present in some `if` conditions all over `background.js`.

File: scripts/background.js [hunk 3]
```diff
@@ -173,235 +171,216 @@ chrome.webRequest.onCompleted.addListener(function(details) {
...
+          name: "similarweb",
+          shouldOpen: event.similarweb,
+          url: url
+        },
+        {
+          name: "tagcloud",
+          shouldOpen: event.tagcloud,
+          url: message.url
+        }
+      ];
```
[the remark this answers is not in the document]
Comment (abhidas17695) [reply scripts/background.js]: I am making changes the way you suggested @vbanos

**Comment by abhidas17695** [comment]
@vbanos I have refactored `openThatContext()` to use the context objects. Please review the changes.

**Pull request merged**